### PR TITLE
Add HTTP Persistent (session reuse) capability

### PR DESCRIPTION
**Pull request by eeyepee**
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

With this update, now the HTTP Request node can support persistent HTTP connection (HTTP session reuse). This is very useful in some cases, for instance, an API that requires subsequent communications through the first initial TCP session.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the mailing list/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: packages/node_modules/@node-red/nodes/core/io/21-httprequest.html [hunk 1]
```diff
@@ -102,6 +107,7 @@
             paytoqs: {value: false},
             url:{value:"",validate:function(v) { return (v.trim().length === 0) || (v.indexOf("://") === -1) || (v.trim().indexOf("http") === 0)} },
             tls: {type:"tls-config",required: false},
+            "http-persistent":{value:true},
```
Comment (knolleary) [inline packages/node_modules/@node-red/nodes/core/io/21-httprequest.html]: Can we rename this to just `persist`? Prefer to keep the node properties as short as is reasonable to minimise the impact on flow file size.

File: packages/node_modules/@node-red/nodes/locales/en-US/messages.json [hunk 2]
```diff
@@ -413,6 +413,7 @@
         "digest": "digest authentication",
         "bearer": "bearer authentication",
         "use-proxy": "Use proxy",
+        "http-persistent": "HTTP Persistent",
```
Comment (knolleary) [inline packages/node_modules/@node-red/nodes/locales/en-US/messages.json]:
```suggestion
        "http-persistent": "Enable connection keep-alive",
```

File: packages/node_modules/@node-red/nodes/core/io/21-httprequest.html (hunk 1 above)
Comment (eeyepee) [reply packages/node_modules/@node-red/nodes/core/io/21-httprequest.html]: Sure, please go ahead.

Comment (dceejay) [reply packages/node_modules/@node-red/nodes/core/io/21-httprequest.html]: (I think he meant you ;-)

Comment (eeyepee) [reply packages/node_modules/@node-red/nodes/core/io/21-httprequest.html]: > 
> 
> Can we rename this to just `persist`? Prefer to keep the node properties as short as is reasonable to minimise the impact on flow file size.

Done

**Comment by eeyepee** [comment]
> Can you also retarget this to the `dev` branch - that is where new features must land.

Moved to the dev branch.

**Comment by eeyepee** [comment]
> 
> 
> Thanks for this.
> 
> There are a couple changes needed before we can merge this.
> 
> Can you also retarget this to the `dev` branch - that is where new features must land.

Done @knolleary.

**Comment by knolleary** [comment]
Thanks - going to merge this and make one important change - you default the value to `true`. The default should be `false` which is existing behaviour of the node.

**Pull request merged**